web/react/components/edit_channel_modal.jsx

```diff
@@ -6,17 +6,25 @@ var AsyncClient = require('../utils/async_client.jsx');
 
 module.exports = React.createClass({
     handleEdit: function(e) {
-        var data = {}
+        var self = this;
```

Comment:
Looks like you're already using ```this``` in those functions anyway, can you replace the ```self``` reference with ```this```